pcdsdevices/gon.py

```diff
@@ -236,23 +255,348 @@ class Kappa(BaseInterface, Device):
 
     prefix_phi : str
         The EPICS base PV of the Kappa stage's phi motor.
-    """
 
+    eta_max_step : int, optional
+        Maximum eta motor step, the largest move eta motor can make without
+        user's confirmation. Defaults to 2.
+
+    kappa_max_step : int, optional
+        Maximum kappa motor step, the largest move kappa motor can make
+        without user's confirmation. Defaults to 2.
+
+    phi_max_step : int, optional
+        Maximum phi motor step, the largest move phi motor can make without
+        user's confirmation. Defaults to 2.
+
+    kappa_ang : number, optional
+        The angle of the kappa motor relative to the eta motor, in degrees.
+        Defaults to 50.
+
+    Notes
+    --------
+    When using the Kappa, it is most convenient to work through the pseudo
+    motors:
+
+    `kappa.e_eta`
+    `kappa.e_chi`
+    `kappa.e_phi`
+
+    Which have the normal motor functionalities (`mv`, `mvr`, `wm`).
+
+    It may be helpful to scan these pseudo motors to find the optimal position,
+    but make sure the step size is small enough so that you don't have to
+    confirm motion on every step.
+
+    Move commands will block the main thread, and pressing ctrl+c will cancel
+    motion.
+
+    The x, y, and z are the sample adjustment motors used to attain center of
+    rotation
+    """
     x = FCpt(IMS, '{self._prefix_x}', kind='normal')
     y = FCpt(IMS, '{self._prefix_y}', kind='normal')
     z = FCpt(IMS, '{self._prefix_z}', kind='normal')
+    # The real (or physical) positioners:
     eta = FCpt(IMS, '{self._prefix_eta}', kind='normal')
     kappa = FCpt(IMS, '{self._prefix_kappa}', kind='normal')
     phi = FCpt(IMS, '{self._prefix_phi}', kind='normal')
+    # The pseudo positioner axes:
+    e_eta = FCpt(PseudoSingleInterface, kind='normal', name='gon_kappa_e_eta')
+    e_chi = FCpt(PseudoSingleInterface, kind='normal', name='gon_kappa_e_chi')
+    e_phi = FCpt(PseudoSingleInterface, kind='normal', name='gon_kappa_e_phi')
 
     tab_component_names = True
 
-    def __init__(self, *, name, prefix_x, prefix_y, prefix_z, prefix_eta,
-                 prefix_kappa, prefix_phi, **kwargs):
+    tab_whitelist = ['stop', 'wait', 'k_to_e', 'e_to_k', 'mv_e_eta',
+                     'mv_e_chi', 'mv_e_phi', 'check_motor_step']
+
+    def __init__(self, *, name, prefix_x, prefix_y, prefix_z,
+                 prefix_eta, prefix_kappa, prefix_phi, eta_max_step=2,
+                 kappa_max_step=2, phi_max_step=2, kappa_ang=50, **kwargs):
         self._prefix_x = prefix_x
         self._prefix_y = prefix_y
         self._prefix_z = prefix_z
         self._prefix_eta = prefix_eta
         self._prefix_kappa = prefix_kappa
         self._prefix_phi = prefix_phi
+        self.eta_max_step = eta_max_step
+        self.kappa_max_step = kappa_max_step
+        self.phi_max_step = phi_max_step
+        self.kappa_ang = kappa_ang
         super().__init__('', name=name, **kwargs)
+
+    def stop(self):
```

Comment:
I added the `stop()` and `wait()` methods directly from the old code, but we might not need them here?
At least I might not need the `stop()` method - but I am not sure

Reply:
I don't think they are strictly necessary, but I think they are really good to have!
If someone is doing `kappa.move` but then changes their mind, they might try `kappa.stop`, and stopping all the motors at once is what they expect to happen.

Reply:
ok that sounds good - i'll keep them here then - thank you